### ISSUE TITLE
Nextcloud 16 won't connect (No 'Authorization: Bearer' header found.)

### ISSUE THREAD

**Issue by atoav** [laurent22/joplin#2219]
When I try to setup Syncing with my own nextcloud instance from my Windows Client, the connection fails with the following Error:
```
Error. Please check that URL, username, password, etc. are correct and that the sync target is accessible. The reported error was:
PROPFIND : Username or password was incorrect, No 'Authorization: Bearer' header found. Either the client didn't send one, or the server is mis-configured (Exception Sabre\DAV\Exception\NotAuthenticated) (401): <?xml version="1.0" encoding="utf-8"?> <d:error xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns"> <s:exception>Sabre\DAV\Exception\NotAuthenticated</s:exception> <s:message>Username or password was incorrect, No 'Authorization: Bearer' header found. Either the client didn't send one, or the server is mis-configured</s:message> </d:error> (Code 401)
```

The credentials come directly from my password manager that I also use for logins into the web interface of nextcloud, so there is no way they are wrong (I additionally tried to type it by hand, with no real success). The URL points to an existing directory on the cloud instance and I tripplechecked that one as well (`https://cloud.atoav.com/remote.php/webdav/Joplin`). My certificate is not self signed.

![nextcloud](https://user-images.githubusercontent.com/6620266/71379227-d7a8b200-25ca-11ea-949b-4dcafb15ac22.png)

The specivity of the message leads me to believe this could be a bug (maybe due to changes how nextclouds wants WebDAV Clients to authenticate?). I didn't do much with my Nextcloud Installation so I would be surprised if this had anything to do with my Server...

## Environment

Joplin version: 1.0.175 (Build 8c1d13b3)
Platform:  Windows 10 Pro
OS specifcs: 10.0.18363 Build 18363 (x64)
Nextcloud Version: 16.0.5.

## Steps To Reproduce

1. Enter username, password and URL
2. Click the Button "Check synchronisation configuration"
3. After a few seconds the above-mentioned Error message shows up.

## Logfile

There is nothing showing up in the logfile – in my judgment the message above should be enough

**Comment by tessus** [comment]
Do you have 2FA activated on Nextcloud? If yes, please open a topic in the forum.

**Comment by atoav** [comment]
> Do you have 2FA activated on Nextcloud?

No, it is actually deactivated

**Comment by tessus** [comment]
Hmm, many people (including myself) use Joplin with Nextcloud. This is the first time I've heard of this error. I can't reproduce the problem.

I'd still suggest to post this to the forum. Maybe somebody has seen this before. The forum is read by a lot more people. The tracker on gh is mainly for Laurent and other devs.

I suspect it's a config problem. Either with Nextcloud, web server, reverse proxy, docker image, ....

**Comment by atoav** [comment]
Thanks for the advice I crossposted [to this topic](https://discourse.joplinapp.org/t/nextcloud-16-wont-connect-no-authorization-bearer-header-found/4842).

**Comment by tessus** [comment]
Yep, I saw it. Let's hope somebody has a solution. I googled the error and I came across a few comments that the authorization header was not properly propagated to the nextcoud system (e.g. from Apache httpd or HAproxy).
I have run tests with Nextcloud 15 and 16 and could not reproduce the issue.
I'm currently pressed for time, otherwise I would have deliberatly introduced misconfigurations to elicit above mentioned error somehow.

**Comment by atoav** [comment]
My nextcloud instance runs behind an (managed) Apache, so I am not completely in control of that Server – might have something to do with that.

**Comment by atoav** [comment]
Thanks for the hint. [This issue in the Nextcloud repo](https://github.com/nextcloud/server/issues/5512) seems to back your theory. I asked the support at my hoster to open a ticket and will post any conclusions here. Right now I can't say 100% certain that it is a Apache issue, but if it is, I will close this.

**Comment by tessus** [comment]
Please post the solution and info to the forum - in case you will have found one. If somebody runs into the same issue in the future, they will have a solution.
Almost noone ever searches gh issues.

**Comment by atoav** [comment]
Okay found a solution.

What I did was to use _my own_ username and password, while in fact I should have created a _device password_ for Joplin and used that one instead. That means I had to go to `Settings → Personal: Security → Devices & Sessions` and create a new App name and password there for Joplin.

Once I used this newly generated password from within Joplin it worked.

This is a bit of a emberassing error, but I might not be the only one who gets this wrong : )
